### DISCUSSION TITLE
iOS Device: self signed certificate in certificate chain

### DISCUSSION THREAD

**Discussion by aarizhov** [baresip/baresip#2742] [Q&A] [answered]
I build baresip with OpenSSL for iOS (simulator and device)

on server side I have WSS

for simulator all works fine, connection success, calls works fine, cool 

BUT

for ios devices I have this issue:

[;mwebsock: connecting to 'wss://<myhost>:443/'   // I hide my host in this line 

[31mtls: verify_handler: err          = 19
[;m[31mtls: verify_handler: error_string = self signed certificate in certificate chain
[;m[31mtls: verify_handler: depth        = 3

Any suggestions, please ?

**Comment by aarizhov** [comment]
ok found issue, but still need help 

**** for ios device ****

ua: adding SIP CA file: /etc/ssl/cert.pem
ua: adding SIP CA path: 
[31mtls: Can't read CA file: /etc/ssl/cert.pem
[;m[31mua: tls_add_ca() failed: No such file or directory
[;m[31mtls: Can't read CA file: /etc/ssl/cert.pem
[;m[31mua: wss tls_add_ca() failed: No such file or directory

**** for simulator  **** 

ua: adding SIP CA file: /etc/ssl/cert.pem
ua: adding SIP CA path: 

for simulator no errors ...


How resolve it for ios device ?

**Comment by sreimers** [comment, marked as answer]
You can disable with `sip_verify_server no` or the more secure way is to deliver a valid cacert.pem like provided by https://curl.se/docs/caextract.html within your app (and configure with `sip_cafile`).

**Reply by aarizhov** [reply]
I just added valid .pem file to project and it's work now for me, thanks
